### ISSUE TITLE
Wallaby fails with `vm.createContext`

### ISSUE THREAD

**Issue by urish** [wallabyjs/public#1271]
### Issue description or question

I have a test suite for a Chrome extension. The test suite uses the Jest framework, thus executing inside Node. The code uses the `vm.createContext()` method to simulate the Chrome extension environment, where there are multiple scripts each running in a separate context and each having a different global object. This works great in Jest, however, with Wallaby.js, I get an error saying `$_$wp` is not defined for the code running inside the created context. I believe this has something to do with the way Wallaby.js instruments the code for coverage.

The extension is open source, and you can find it here: https://github.com/urish/web-bluetooth-polyfill

Basically, if you run `npm test` you will see all tests pass, but when running through Wallaby.js, I get the `$_$wp` not defined error.

### Wallaby.js configuration file

```javascript
module.exports = function (wallaby) {
    return {
        files: [
            'extension/*.js',
            'tests/*.js',
            { pattern: 'tests/*.spec.js', ignore: true },
        ],

        tests: ['tests/*.spec.js'],

        env: {
            type: 'node',
            runner: 'node'
        },

        testFramework: 'jest'
    };
};
```

### Code editor or IDE name and version

Visual Studio Code v1.15.1

### OS name and version

Windows 10

**Comment by ArtemGovorov** [comment]
Thanks for providing the repo link.

Wallaby needs few globals in order to function and `vm` doesn't pass anything that's not whitelisted to the sandbox.

The easiest way to make it work is just to use the `setup` function to patch vm function you're using and pass the required globals:

```javascript
        setup: () => {
            if (!global._vmPatched) {
                const vm = require('vm');
                const createContext = vm.createContext;
                const runInNewContext = vm.Script.prototype.runInNewContext;
                const wallabyGlobals = {$_$wp, $_$wpe, $_$w, $_$wf, $_$wv, $_$tracer};
                vm.createContext = function () {
                    arguments[0] = Object.assign(arguments[0], wallabyGlobals);
                    return createContext.apply(this, arguments);
                };
                vm.Script.prototype.runInNewContext = function () {
                    arguments[0] = Object.assign(arguments[0], wallabyGlobals);
                    return runInNewContext.apply(this, arguments);
                };
                global._vmPatched = true;
            }
        }
```

I have sent you [the PR to make it work](https://github.com/urish/web-bluetooth-polyfill/pull/25).

**Comment by urish** [comment]
Amazing, thanks Artem!

Here is a short story for you:

This week I gave a talk about Testing Angular Code Efficiently in our local Angular meetup. We had about 200 attendees, and at some point I introduced Wallaby, showing how you get instant test results instead of having to wait a few seconds for Karma to do its magic every time. I noticed a few jaw drops in the audience - they were really impressed. So please keep up the good work :-)

**Comment by ArtemGovorov** [comment]
@urish Great, thanks for sharing Uri, you rock!
